operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/ResourceController.java

```diff
@@ -6,6 +6,23 @@
 
 public interface ResourceController<R extends CustomResource> {
 
+  static String getDefaultNameFor(ResourceController controller) {
```

Comment:
Wonder why these methods are needed to be here.  Let's keep the interfaces as clean and relevant as possible! 
I still believe the getName also should be pulled out from this interface, I think it's been missed in the main PR wrt quarkus extension.

Reply:
We don't need to put these methods there, it just that it made sense to me to move them here since they pertain to `ResourceControllers` instead of adding them to a util class that's less discoverable.

Reply:
@psycho-ir why do you think the `getName` method should be pulled? It's needed to be able to configure controllers individually.

Reply:
It let the user set the name in 2 different ways, @Controller.name & overriding the `getName` method and that's what we should strive to avoid.

Reply:
Ah yes, didn't think of that but I assume that someone would need to know what they're doing if they override the default `getName` method… since we provide a default implementation. And if they provide their own implementation, then I guess that's the one that would be used and the annotation would be ignored anyway. I don't think that's something that would be an issue in practice.

Reply:
Well, API design is mainly about the things we let users do and the reason behind them. 
The getName method has been implemented for internal use and is effectively final (no reason to be overridden by the client).

Having that in the interface as the method will lead to other questions like why `finalizerName`, `crdName`, `generationAwareEventProcessing` and so on don't have getter method in interface accordingly if there is a getter method with default implementation why @Controller annotation is getting the `name` field too? and several similar questions.

Reply:
The name is not internal: to be able to configure a controller, you need to be able to identify it. That's the name's purpose. While there are no obvious reasons that I can think of to override the default naming scheme, some people might still want to provide their own way to name a controller independently of what we provide (because for example, they're not using annotations and don't want to incur that impact at runtime). As such, it makes sense (at least to me) that it lives on the `ResourceController` interface. 

The other methods, on the other hand, are part of the controller's configuration, which, again, you can't retrieve without first knowing the name of the controller. These values are not part of what is essential to the controller.

Reply:
In comment, I was talking about the **getName** and not **name**, ofc **name** is not an internal thing, thus the user can set it via @Controller.name. 

**getName** for sure can be added **later on** (it's backward compatible change) to the interface if it's been decided to get controller essential configuration from anywhere other than the controller annotation, AFAIK it's not currently the case and annotation + its crdName property is required.

Reply:
Fixed.